### PR TITLE
Document branching & go.mod update strategies

### PR DESCRIPTION
**Pull request by tallclair**
Fixes https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/455

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,14 @@ You can reach the maintainers of this project at:
 
 Participation in the Kubernetes community is governed by the [Kubernetes Code of Conduct](code-of-conduct.md).
 
+## Releases
+
+To reduce version-skew risk, it is recommended to match apiserver-network-proxy server & client
+minor release versions. With Kubernetes, this means:
+
+- Kubernetes versions up to v1.27: `0.0.X` apiserver-network-proxy branch
+- Kubernetes v1.27+: `0.1.X` apiserver-network-proxy branch
```
Comment (jkh52) [inline README.md:26]: might be worth adding 'master branch' to this entry

File: RELEASE.md [hunk 2]
```diff
@@ -66,3 +66,16 @@ Creating a new release of network proxy involves releasing a new version of the
...
+
+`go.mod` versions should be kept consistent between the apiserver-network-proxy [main module](go.mod),
+and the [konnectivity-client module](/konnectivity-client/go.mod).
+
+Konnectivity-client dependency versions must be compatible
+[Kubernetes go.mod versions](https://github.com/kubernetes/kubernetes/blob/master/go.mod),
+meaning for a given release branch the konnectivity-client versions must not be newer than the
+Kubernetes versions.
+
+In practice, this means that to update go.mod dependencies used by the client, a new Konnectivity
```
Comment (jkh52) [inline RELEASE.md:80]: 'konnectivity-client/go.mod' here might avoid confusion

**Comment by jkh52** [comment]
This doesn't mention how far back we strive to keep 0.0.X compatible. Only with supported k/k versions?

**Comment by jkh52** [comment]
It may be worth also suggesting a command to programmatically check backport-ability, see https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/472

**Comment by ipochi** [comment]
> This doesn't mention how far back we strive to keep 0.0.X compatible. Only with supported k/k versions?

> It may be worth also suggesting a command to programmatically check backport-ability, see #472

This definitely needs clarity as we upgrade our systems to use the new Konnectivity versions, preferably a version matrix.

On a more advanced note perhaps a test framework to check for compatiblity issues with older k/k releases with the newer versions of Konnectivity.

**Comment by ipochi** [comment]
@cheftako friendly ping.

**Comment by ipochi** [comment]
> 

@cheftako friendly ping.

**Comment by ipochi** [comment]
> > This doesn't mention how far back we strive to keep 0.0.X compatible. Only with supported k/k versions?
> 
> > It may be worth also suggesting a command to programmatically check backport-ability, see #472
> 
> This definitely needs clarity as we upgrade our systems to use the new Konnectivity versions, preferably a version matrix.
> 
> On a more advanced note perhaps a test framework to check for compatiblity issues with older k/k releases with the newer versions of Konnectivity.

@cheftako friendly ping.

**Comment by tallclair** [comment]
Comments addressed.

@jkh52 

> It may be worth also suggesting a command to programmatically check backport-ability, see https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/472

I'm not sure exactly what you had in mind here. Maybe you could add it in a follow-up?

File: README.md [hunk 3]
```diff
@@ -22,8 +22,10 @@ Participation in the Kubernetes community is governed by the [Kubernetes Code of
 To reduce version-skew risk, it is recommended to match apiserver-network-proxy server & client
 minor release versions. With Kubernetes, this means:
 
-- Kubernetes versions up to v1.27: `0.0.X` apiserver-network-proxy branch
-- Kubernetes v1.27+: `0.1.X` apiserver-network-proxy branch
+- Kubernetes v1.27+: `0.1.X` tags, `master` branch
+- Kubernetes versions v1.23 to v1.27: `0.0.X` tags, `release-0.0` branch
+- Kubernetes versions up to v1.23: apiserver-network-proxy versions up to `v0.0.30`.
+  Refere to the kubernetes go.mod file for the specific release version.
```
Comment (jkh52) [inline README.md:28]: s/Refere/Refer

**Pull request merged**